internal/controllers/topology/cluster/structuredmerge/serversidepathhelper.go

```diff
@@ -0,0 +1,196 @@
+	var hasChanges, hasSpecChanges bool
+	switch original {
+	case nil:
+		hasChanges, hasSpecChanges = true, true
+	default:
+		hasChanges, hasSpecChanges = dryRunPatch(&hasChangesContext{
+			path:     contract.Path{},
+			fieldsV1: getTopologyManagedFields(original),
+			original: originalUnstructured.Object,
+			modified: modifiedUnstructured.Object,
+		})
+	}
+
+	return &serverSidePatchHelper{
+		client:         c,
+		modified:       modifiedUnstructured,
+		hasChanges:     hasChanges,
+		hasSpecChanges: hasSpecChanges,
+	}, nil
+}
+
+// HasSpecChanges return true if the patch has changes to the spec field.
+func (h *serverSidePatchHelper) HasSpecChanges() bool {
+	return h.hasSpecChanges
+}
+
+// HasChanges return true if the patch has changes.
+func (h *serverSidePatchHelper) HasChanges() bool {
+	return h.hasChanges
+}
+
+// Patch will server side apply the current intent (the modified object.
+func (h *serverSidePatchHelper) Patch(ctx context.Context) error {
+	if !h.HasChanges() {
+		return nil
+	}
+
+	log := ctrl.LoggerFrom(ctx)
+	log.V(5).Info("Patching object", "Intent", h.modified)
+
+	options := []client.PatchOption{
+		client.FieldOwner(topologyManagerName),
+		// NOTE: we are using force ownership so in case of conflicts the topology controller
+		// overwrite values and become sole manager.
+		client.ForceOwnership,
+	}
+	return h.client.Patch(ctx, h.modified, client.Apply, options...)
+}
+
+// cleanupLegacyManagedFields cleanups managed filed management in place before introducing SSA.
+// NOTE: this operation can trigger a machine rollout, but this is considered acceptable given that ClusterClass is still alpha
+// and SSA adoption align the topology controller with K8s recommended solution for many controllers authoring the same object.
+func cleanupLegacyManagedFields(obj *unstructured.Unstructured, c client.Client) error {
+	base := obj.DeepCopyObject().(*unstructured.Unstructured)
+
+	// Remove the topology.cluster.x-k8s.io/managed-field-paths annotation
+	annotations := obj.GetAnnotations()
+	delete(annotations, clusterv1.ClusterTopologyManagedFieldsAnnotation)
+	obj.SetAnnotations(annotations)
+
+	// Remove managedFieldEntry for manager=manager and operation=update to prevent having two managers holding values set by the topology controller.
+	originalManagedFields := obj.GetManagedFields()
+	managedFields := make([]metav1.ManagedFieldsEntry, 0, len(originalManagedFields))
+	for i := range originalManagedFields {
+		if originalManagedFields[i].Manager == "manager" &&
+			originalManagedFields[i].Operation == metav1.ManagedFieldsOperationUpdate {
+			continue
+		}
+		managedFields = append(managedFields, originalManagedFields[i])
+	}
+
+	// Add a seeding managedFieldEntry for SSA executed by the management controller, to prevent SSA to create ""
+	// NOTE: fieldV1Map cannot be empty, so we add metadata.name which will be cleaned up at the first SSA patch.
+	fieldV1Map := map[string]interface{}{
```

Comment:
metadata.name is usually not tracked as a managed field, it acts as a placeholder for the actual managed fields and it will be removed at the first reconcile.
if we drop the managed field entirely server-side apply automatically rebuilds/infer an initial one, and this leads to co-ownership for all the fields

Reply:
improved comment
the doc you are referring describes how you can change managed fields, instead, we are concerned about what happen when you SSA on a object without managedFields; in this case, SSA automatically creates an initial managedField with operation=update, kind of rebuilding the missing history of the existing object, and this leads to co-ownership.